src/pages/api/umamiService.ts

```
@@ -0,0 +1,134 @@
+const WEBSITE_ID = process.env.UMAMI_WEBSITE_ID;
+const REQUEST_TIMEOUT = 5000;
+const RATE_LIMIT_INTERVAL = 60000;
+const MAX_REQUESTS = 5;
+
+// Rate limiting variables
+let requestCount = 0;
+let lastRequestTime = Date.now();
+
+// Check for valid Date
+function isValidDate(date: string): boolean {
+  return !isNaN(Date.parse(date));
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)